server/lib/amplitude.js

```diff
@@ -136,7 +138,7 @@ module.exports = receiveEvent;
 
 function receiveEvent (event, request, data) {
   if (! event || ! data) {
-    return;
+    return P.resolve();
```

Comment:
I don't think so. The intent, before this function became async with the introduction of geolocation, was that it would never throw regardless of the arguments. It feels like the natural continuation of that policy is not to reject either.

But anyway, nothing depends on the returned promise, the function is invoked fire-and-forget. The only reason I'm returning the promise now is so that the tests can wait for everything to finish before asserting. So if it did reject, we'd be introducing the possibility of `Unhandled promise rejection` errors in the production code, which definitely seems wrong.

Reply:
On a side note, `geodb()` should not be asynchronous. It's a memory access.

Reply:
Oh wow, that's a great catch, thanks @jrgm. I assumed it was async because `fxa-geodb` returns a promise, but I just checked the source code in that repo and it seems to return a promise for no good reason. I'll file an issue to change it (and bump the major rev) over there.